ttnn/cpp/ttnn/operations/moreh/moreh_dot_op_backward/device/moreh_dot_backward_program_factory.cpp

```diff
@@ -0,0 +1,167 @@
+        (std::uint32_t)dst0_is_dram,
+        (std::uint32_t)dst1_is_dram,
+    };
+
+    const auto reader_kernel_file =
+        "ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/moreh_dot_backward/kernels/reader_moreh_dot_backward.cpp";
+    const auto writer_kernel_file =
+        "ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/moreh_dot_backward/kernels/writer_moreh_dot_backward.cpp";
+
+    const auto reader_kernel_id = tt::operations::primary::CreateReadKernel(program, reader_kernel_file, core, reader_compile_time_args);
+    const auto writer_kernel_id = tt::operations::primary::CreateWriteKernel(program, writer_kernel_file, core, writer_compile_time_args);
+
+    vector<uint32_t> compute_kernel_args = {};
+    std::map<string, string> compute_defines;
+
+    const auto compute_kernel_file = "ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/moreh_dot_backward/kernels/moreh_dot_backward.cpp";
+    const auto compute_kernel_id =
+        tt::operations::primary::CreateComputeKernel(program, compute_kernel_file, {core, core_num, compute_kernel_args}, compute_defines);
+
+    SetRuntimeArgs(
+        program,
+        reader_kernel_id,
+        core,
+        {(std::uint32_t)has_input_grad,
+         (std::uint32_t)has_other_grad,
+         src0_buffer->address(),
+         src1_buffer->address(),
+         src2_buffer->address(),
+         num_tiles,
+         0});
+
+    SetRuntimeArgs(
+        program, compute_kernel_id, core, {(std::uint32_t)has_input_grad, (std::uint32_t)has_other_grad, num_tiles});
+
+    SetRuntimeArgs(
+        program,
+        writer_kernel_id,
+        core,
+        {(std::uint32_t)has_input_grad, (std::uint32_t)has_other_grad, dst0_address, dst1_address, num_tiles, 0});
+
+    return {
+        std::move(program),
+        {.unary_reader_kernel_id = reader_kernel_id, .unary_writer_kernel_id = writer_kernel_id}};
+}
+
+void MorehDotBackwardOperation::SingleCore::override_runtime_arguments(
+    cached_program_t& cached_program,
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& tensor_return_value) {
+    auto& program = cached_program.program;
+    auto& unary_reader_kernel_id = cached_program.shared_variables.unary_reader_kernel_id;
+    auto& unary_writer_kernel_id = cached_program.shared_variables.unary_writer_kernel_id;
+
+    const auto& output_grad_buffer = tensor_args.output_grad.buffer();
+    const auto& input_buffer = tensor_args.input.buffer();
+    const auto& other_buffer = tensor_args.other.buffer();
+    const auto& input_grad_buffer = tensor_return_value.at(0).buffer();
+    const auto& other_grad_buffer = tensor_return_value.at(1).buffer();
+
+    {
+        auto& runtime_args = tt::tt_metal::GetRuntimeArgs(program, unary_reader_kernel_id, CoreCoord{0, 0});
+        runtime_args[0] = output_grad_buffer->address();
+        runtime_args[1] = input_buffer->address();
+        runtime_args[2] = other_buffer->address();
```

Comment:
```suggestion
        runtime_args[2] = output_grad_buffer->address();
        runtime_args[3] = input_buffer->address();
        runtime_args[4] = other_buffer->address();
```

Please change this part to 2, 3, and 4. After checking reader_moreh_dot_backward.cpp, the indices where the tensor addresses go are 2, 3, and 4.

```C++
    uint32_t has_input_grad = get_arg_val<uint32_t>(0);
    uint32_t has_other_grad = get_arg_val<uint32_t>(1);
    uint32_t src0_addr = get_arg_val<uint32_t>(2);
    uint32_t src1_addr = get_arg_val<uint32_t>(3);
    uint32_t src2_addr = get_arg_val<uint32_t>(4);
```

https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/deprecated/tt_dnn/op_library/moreh_dot_backward/kernels/reader_moreh_dot_backward.cpp#L10

Reply:
I will approve once this is fixed

Reply:
Oh my bad, I didn't pay attention to the reader and writer. I fixed that